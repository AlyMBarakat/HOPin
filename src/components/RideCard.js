import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
} from 'react-native';



const RideCard = ({ name = "Ahmed Khalid", car = "Mitsubishi Lancer 2017", road = "Desert Road", time = "9:30 AM", seatsLeft = "4", }) => {
    return (
        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
            <Text style={{ marginRight: 7, fontSize: 14, color: "#fff", fontWeight: "bold", width: 70 }}>
                {time}
            </Text>
            <View style={styles.container}>
                <Text style={{ fontSize: 20, color: "#fff", fontWeight: "bold" }}>{name}</Text>
                <Text style={{ fontSize: 15, color: "#3C796D", fontWeight: "bold" }}>{car}</Text>
                <Text style={{ fontSize: 15, color: "#3C796D", fontWeight: "bold" }}>Seats left: {seatsLeft}</Text>
                <View style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center' }}>
                    <Text style={{ fontSize: 15, color: "#3C796D", fontWeight: "bold" }}>{road}</Text>
                    {/* HOPin Button */}
                    <TouchableOpacity style={styles.hop}>
                        <Text style={{ fontSize: 14, color: "#fff", fontWeight: 'bold' }}>HOP ðŸ¤™</Text>
                    </TouchableOpacity>
                </View>
            </View>
        </View>
    )
}


const styles = StyleSheet.create({
    container: {
        backgroundColor: "#72B5A8",
        padding: 17,
        borderRadius: 5,
        marginBottom: 10,
        flex: 1,
    },
    hop: {
        backgroundColor: "#3C796D",
        padding: 15,
        borderRadius: 5,
    }

})

export default RideCard;