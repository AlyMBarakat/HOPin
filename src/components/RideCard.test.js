import React from 'react';
import renderer from 'react-test-renderer';
import RideCard from './RideCard';

test('renders RideCard correctly', () => {
  const tree = renderer
    .create(
      <RideCard
        name="Ahmed Khalid"
        car="Mitsubishi Lancer 2017"
        road="Desert Road"
        time="9:30 AM"
        seatsLeft="4"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
