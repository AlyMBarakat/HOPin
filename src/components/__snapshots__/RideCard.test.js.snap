// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RideCard correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginRight": 7,
        "width": 70,
      }
    }
  >
    9:30 AM
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#72B5A8",
        "borderRadius": 5,
        "flex": 1,
        "marginBottom": 10,
        "padding": 17,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Ahmed Khalid
    </Text>
    <Text
      style={
        Object {
          "color": "#3C796D",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Mitsubishi Lancer 2017
    </Text>
    <Text
      style={
        Object {
          "color": "#3C796D",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Seats left: 
      4
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3C796D",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        Desert Road
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3C796D",
            "borderRadius": 5,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          HOP ðŸ¤™
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders date and time pickers correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginRight": 7,
        "width": 70,
      }
    }
  >
    9:30 AM
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#72B5A8",
        "borderRadius": 5,
        "flex": 1,
        "marginBottom": 10,
        "padding": 17,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Ahmed Khalid
    </Text>
    <Text
      style={
        Object {
          "color": "#3C796D",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Mitsubishi Lancer 2017
    </Text>
    <Text
      style={
        Object {
          "color": "#3C796D",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Seats left: 
      4
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3C796D",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        Desert Road
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#3C796D",
            "borderRadius": 5,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          HOP ðŸ¤™
        </Text>
      </View>
    </View>
  </View>
</View>
`;
